### ISSUE TITLE
[REQ] This is a Intersight Feature request to enable the provider to send size and to for pools

### ISSUE THREAD

**Issue by scotttyso** [CiscoDevNet/terraform-provider-intersight#178]
**Is your feature request related to a problem? Please describe.**

Currently for all the pools you can send the from and size or to attributes during an initial plan.  But subsequent plans will show a change to the resource with both the size and to attributes, which makes it look like there is a change.

Please enable the ability in the Intersight API to send from, size, and to in an initial plan.

Example Below

```hcl
variable "mac_pools" {
  default = {
    default = { # The Pool Name will be {each.key}.  In this case it would be default if left like this.
      assignment_order = "default"
      description      = ""
      organization     = "default"
      tags             = []
      mac_blocks = {
        default = {
          from = "00:25:B5:0A:00:00"
          size = 1000
          to   = "00:25:B5:0A:03:E7"
        }
      }
    }
  }
  description = <<-EOT
  key - Name of the MAC Pool.
  * Assignment order decides the order in which the next identifier is allocated.
    - default - (Default) Assignment order is decided by the system.
    - sequential - Identifiers are assigned in a sequential order.
  * description - Description to Assign to the Pool.
  * mac_blocks - Map of Addresses to Assign to the Pool.
    - from - Staring MAC Address.  An Example is "00:25:B5:0A:00:00".
    - size - Size of MAC Address Pool.  An Example is 1000.
    - to - Ending MAC Address.  An Example is "00:25:B5:0A:03:E7".
  * organization - Name of the Intersight Organization to assign this pool to.  Default is default.
    - https://intersight.com/an/settings/organizations/
  * tags - List of Key/Value Pairs to Assign as Attributes to the Pool.
  EOT
  type = map(object(
    {
      assignment_order = optional(string)
      description      = optional(string)
      mac_blocks = optional(map(object(
        {
          from = string
          size = optional(number)
          to   = optional(string)
        }
      )))
      organization = optional(string)
      tags         = optional(list(map(string)))
    }
  ))
}


#____________________________________________________________
#
# Intersight Pools Module
# GUI Location: Pools > Create Pool
#____________________________________________________________

#______________________________________________
#
# MAC Pools
#______________________________________________

resource "intersight_macpool_pool" "mac_pools" {
  depends_on = [
    local.org_moids
  ]
  for_each         = local.mac_pools
  assignment_order = each.value.assignment_order
  description      = each.value.description != "" ? each.value.description : "${each.key} MAC Pool"
  name             = each.key
  dynamic "mac_blocks" {
    for_each = each.value.mac_blocks
    content {
      object_type = "macpool.Block"
      from        = mac_blocks.value.from
      size        = mac_blocks.value.size != null ? tonumber(mac_blocks.value.size) : null
      to          = mac_blocks.value.to != null ? mac_blocks.value.to : null
    }
  }
  organization {
    moid        = local.org_moids[each.value.organization].moid
    object_type = "organization.Organization"
  }
  dynamic "tags" {
    for_each = length(each.value.tags) > 0 ? each.value.tags : local.tags
    content {
      key   = tags.value.key
      value = tags.value.value
    }
  }
}

locals {
  mac_pools = {
    for k, v in var.mac_pools : k => {
      assignment_order = v.assignment_order != null ? v.assignment_order : "default"
      description      = v.description != null ? v.description : ""
      mac_blocks       = v.mac_blocks != null ? v.mac_blocks : {}
      organization     = v.organization != null ? v.organization : "default"
      tags             = v.tags != null ? v.tags : []
    }
  }
}
```

Today I have to comment out the to field on the initial run and then uncomment it on subsequent runs

```hcl
mac_pools = {
  "MGMT-A" = {
    mac_blocks = {
      "1" = {
        from = "00:25:B5:00:1A:00"
        size = 255
        # to   = "00:25:B5:00:1A:FE"
      },
    }
    organization = "Core"
    tags         = []
  }
}
```
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Include terraform configurations supporting the above request

**Comment by jerewill-cisco** [comment]
You can workaround this behavior by using a lifecycle block to ignore the drift that this causes in mac_blocks.
 
```
resource "intersight_macpool_pool" "cisco_af_1" {
  name = "cisco_af_1"

  mac_blocks {
    from = "00:25:B5:AF:10:00"
    size = 1000
  }

  lifecycle {
    ignore_changes = [mac_blocks]
  }

}
```

**Comment by scotttyso** [comment]
that works if you never plan to add another block.  That isn't a valid solution to the request.

**Comment by scotttyso** [comment]
Unfortunately for my use case that doesn't address the ability for a user to add an additional block in the future.

**Comment by cgascoig** [comment]
I've just been looking at the same issue for the IKS provider - you can be more specific with your ignore_changes:
```
  lifecycle {
    ignore_changes = [  
      ip_v4_blocks[0].to
    ]
  }
```

This works but only if you know how many blocks there are (using `ip_v4_blocks[*].to` would be ideal but isn't supported by TF).

Ultimately I would like to see this addressed in the provider - either `size` or `to` could be changed to a readonly / computed property so that there is only 1 way to set these blocks.

**Comment by jerewill-cisco** [comment]
> Ultimately I would like to see this addressed in the provider - either `size` or `to` could be changed to a readonly / computed property so that there is only 1 way to set these blocks. 

If we're voting one of the properties out of the provider, I think that "to" is the one to remove.  Size makes more sense to me, being easier to work with using arithmetic operators.

**Comment by cgascoig** [comment]
I'm not particularly bothered which one was to go (personally I think `to` is safer to use because there is less chance of confusion/off-by-one errors in operator's mental arithmetic) but `size` is definitely used in more examples and is therefore probably much more commonly used today. 

Also, I was just playing with ignore_changes a bit more - although you can't use `[*]` in the lifecycle meta attribute (you can only use literals), it doesn't matter if those literals don't exist so it seems like you can do the following to ignore changes on the `to` attribute for the first 5 blocks:
```
  lifecycle {
    ignore_changes = [  
      ip_v4_blocks[0].to,
      ip_v4_blocks[1].to,
      ip_v4_blocks[2].to,
      ip_v4_blocks[3].to,
      ip_v4_blocks[4].to,
    ]
  }
```

**Comment by jerewill-cisco** [comment]
I think `to` and `size` are probably pretty similar in terms of utility for IP pools.  But it's cumbersome to work with hexadecimals in Terraform for WWN and MAC pools.

**Comment by scotttyso** [comment]
Note: This has to apply to all automation strategies not just Terraform as the provider is auto generated at the same time as the SDK and other services.

When I had previously asked for the to/size to be exposed it was because all the pools were inconsistent.  some used size and some used to.  

Whatever is chosen we just need one to either only have one exposed and for the other field to not come back after the initial plan like it is doing today or expose both fields for initial apply.

I am okay with whatever the fix is but it just needs to be fixed.  

Note: this is a much lower priority than fixing policy buckets.  That would be priority 1

**Comment by meghana-rajashekar** [comment]
#### Fixed with latest provider release: v1.0.26
TF now ignores the change in "to" or "size" property in subsequent plans when either of them is passed in the config.
